src/components/create-vm-dialog/createVmDialog.jsx

```diff
@@ -748,20 +752,88 @@ const UsersConfigurationRow = ({
+});
+
+const SshKeysRow = ({
+    id, item, onChange, idx, removeitem,
+}) => {
+    const [keyObject, setKeyObject] = useState();
+
+    return (
+        <Grid id={id} key={id}>
+            <GridItem span={11}>
+                <FormGroup label={_("Public key")} fieldId={id + "-public-key"}>
+                    {keyObject
+                        ? <FlexItem id="validated">
+                            <strong>{keyObject.comment}</strong>
+                            <span>{keyObject.comment ? " - " + keyObject.type : keyObject.type}</span>
+                            <div>{keyObject.data}</div>
+                        </FlexItem>
+                        : <TextArea value={item.value || ""}
+                            aria-label={_("Public SSH key")}
+                            onChange={(_, value) => {
+                                onChange(idx, "value", value);
+                                validateKey(value, setKeyObject);
+                            }}
+                            rows="3" />
+                    }
+                </FormGroup>
+            </GridItem>
+            <GridItem span={1} className="pf-m-1-col-on-md remove-button-group">
+                <Button variant='plain'
+                    className="btn-close"
+                    id={id + "-btn-close"}
+                    isSmall
+                    aria-label={_("Remove item")}
+                    icon={<TrashIcon />}
+                    onClick={() => removeitem(idx)} />
+            </GridItem>
+        </Grid>
+    );
+};
+
 const CloudInitOptionsRow = ({
     onValueChanged,
     rootPassword,
     userLogin, userPassword,
     validationFailed,
 }) => {
     return (
-        <UsersConfigurationRow rootPassword={rootPassword}
-                               rootPasswordLabelInfo={_("Leave the password blank if you do not wish to set a root password")}
-                               showUserFields
-                               userLogin={userLogin}
-                               userPassword={userPassword}
-                               validationFailed={validationFailed}
-                               onValueChanged={onValueChanged} />
+        <>
+            <UsersConfigurationRow rootPassword={rootPassword}
+                                   rootPasswordLabelInfo={_("Leave the password blank if you do not wish to set a root password")}
+                                   showUserFields
+                                   userLogin={userLogin}
+                                   userPassword={userPassword}
+                                   validationFailed={validationFailed}
+                                   onValueChanged={onValueChanged} />
+            <DynamicListForm id="create-vm-dialog-ssh-key"
+                emptyStateString={_("No SSH keys specified")}
+                label={_("SSH keys")}
+                actionLabel={_("Add SSH keys")}
+                onChange={value => onValueChanged('sshKeys', value)}
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1150-20231003-232224-a08f566d-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)